public/jsx/views/card-listing.jsx

```diff
@@ -17,45 +17,45 @@ module.exports = React.createClass({
 
   mixins: [Navigation],
 
-  getInitialState: function() {
-    return {
-      cards: null,
-    };
+  contextTypes: {
+    router: React.PropTypes.func,
   },
 
-  componentDidMount: function() {
-    var user = UserStore.getSignedInUser();
-    console.log('got logged in user:', user);
-
-    if (user.payment_methods.length) {
-      console.log('Card data found, showing card form');
-      this.setState({cards: user.payment_methods});
-    } else {
-      console.log('No card data found, showing card form');
-      this.transitionTo('card-form');
+  selectData: function(state) {
+    console.log('card-listing: selectData() firing with', state);
+    return {
+      user: state.user,
     }
   },
 
-  contextTypes: {
-    router: React.PropTypes.func,
-  },
-
   render: function() {
-    if (this.state.cards) {
-      return (
-        <div className="card-listing">
-          <ProductDetail productId={this.props.productId} />
-          <CardChoice
-            cards={this.state.cards}
-            productId={this.props.productId}
-          />
-        <Link
-            className="card-add bottom-link"
-            to="card-form">{gettext('Add new credit card')}</Link>
-        </div>
-      );
-    } else {
-      return <Spinner text={gettext('Loading')}/>;
-    }
+    var component = this;
+    return (
+      <Connector select={this.selectData}>
+        {function(result) {
+          if (!result.user.email) {
+            return <Spinner text={gettext('Loading')}/>;
```

Comment:
It's sort of weird. Because this component is listening (via the Connector) to user state change, it's possible that this might get rendered. It's unlikely but possible. Most likely the login view will be rendered and a login occur and the card listing view will get reloaded.

I think this is really where trying to use the router starts to break down. I have another patch in the works that gets rid of the router and takes a more data/state driven approach to rendering views.